### PR TITLE
fix(sdk): remove ts limitation on custom fonts + some minimal e2e tests for the fonts

### PR DESCRIPTION
**Pull request by npretto**
Implementes some tests for https://github.com/metabase/metabase/issues/47624 + some fixes of things I was writing the tests for but didn't actually work 😅 

## Changes:

- the redux store is now not a singleton, read comments on the files for the reason
- fix to use Lato as default when we can't connect to metabase
- fix/product change to allow any font to be passed to theme.fontFamily (this is pending [this slack conversation](https://metaboat.slack.com/archives/C063Q3F1HPF/p1727340304714819)

## How to verify:
- on one terminal run `IS_EMBEDDING_SDK=true yarn storybook`
- on another run cypress with yarn `test-cypress-open-sdk` and run `metabase-sdk-styles-tests.cy.spec.ts`

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ module.exports = {
       }),
       new webpack.EnvironmentPlugin({
         EMBEDDING_SDK_VERSION,
-        IS_EMBEDDING_SDK_BUILD: isEmbeddingSDK,
+        IS_EMBEDDING_SDK: isEmbeddingSDK,
```
Comment (npretto) [inline .storybook/main.js:58]: we had both "IS_EMBEDDING_SDK" and "IS_EMBEDDING_SDK_BUILD" which was confusing, I changed so we use only one

File: .storybook/preview.tsx [hunk 2]
```diff
@@ -1,10 +1,14 @@
 import React from "react";
 import { ThemeProvider } from "metabase/ui";
 
-import "metabase/css/core/index.css";
-import "metabase/css/vendor.css";
-import "metabase/css/index.module.css";
-import "metabase/lib/dayjs";
+const isEmbeddingSDK = process.env.IS_EMBEDDING_SDK === "true";
+
+if (!isEmbeddingSDK) {
```
Comment (npretto) [inline .storybook/preview.tsx:6]: for the sake of testing, we should be on a blank page with no css, otherwise we can't test if the provider is actually setting the css

File: .storybook/preview.tsx [hunk 3]
```diff
@@ -35,17 +39,19 @@ const globalStyles = css`
...
-      <ThemeProvider>
-        <Global styles={globalStyles} />
-        <CssVariables />
-        {renderStory()}
-      </ThemeProvider>
-    </EmotionCacheProvider>
-  ),
-];
+export const decorators = isEmbeddingSDK
+  ? [] // No decorators for Embedding SDK stories, as we want to simulate real use cases
```
Comment (npretto) [inline .storybook/preview.tsx:43]: Same as before, for the SDK we need to be on a clean state, simulating the customer app

File: e2e/test/scenarios/embedding-sdk/metabase-sdk-styles-tests.cy.spec.ts [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    restore();
+    cy.signInAsAdmin();
+    setTokenFeatures("all");
+    setupJwt();
+    cy.signOut();
+  });
+
+  describe("style leaking", () => {
+    it("[success scenario] should use the default fonts outside of our components, and Lato on our components", () => {
+      visitFullAppEmbeddingUrl({
```
Comment (npretto) [inline e2e/test/scenarios/embedding-sdk/metabase-sdk-styles-tests.cy.spec.ts:36]: we should make a helper specific to the SDK, but let's do it on another PR

File: enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx [hunk 5]
```diff
@@ -95,9 +95,12 @@ export const MetabaseProviderInternal = ({
   );
 };
 
-export const MetabaseProvider = memo(function MetabaseProvider(
-  props: MetabaseProviderProps,
-) {
+export const MetabaseProvider = memo(function MetabaseProvider({
+  // @ts-expect-error -- we don't want to expose the store prop
+  // eslint-disable-next-line react/prop-types
+  store = getSdkStore(),
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/components/public/MetabaseProvider.tsx:101]: This makes it so the store is not a singleton, so on storybook it's a fresh instance on each story.
Otherwise the auth state was shared across stories making the order in which you opened them change the behaviour 😅

File: enterprise/frontend/src/embedding-sdk/store/index.ts [hunk 6]
```diff
@@ -18,15 +18,16 @@ export const sdkReducers = {
...
-  embed: {
-    options: {},
-    isEmbeddingSdk: true,
-  },
-  app: {
-    isDndAvailable: false,
-  },
-}) as unknown as Store<SdkStoreState, AnyAction>;
+export const getSdkStore = () =>
+  getStore(sdkReducers, null, {
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/store/index.ts:22]: Needed for the change in MetabaseProvider

File: enterprise/frontend/src/embedding-sdk/test/CommonSdkStoryWrapper.tsx [hunk 7]
```diff
@@ -11,7 +11,10 @@ const METABASE_JWT_SHARED_SECRET =
 
 const secret = new TextEncoder().encode(METABASE_JWT_SHARED_SECRET);
 
-const DEFAULT_CONFIG: SDKConfig = {
+/**
+ * SDK config that signs the jwt on the FE
+ */
+export const storybookSdkDefaultConfig: SDKConfig = {
```
Comment (npretto) [inline enterprise/frontend/src/embedding-sdk/test/CommonSdkStoryWrapper.tsx:17]: exporting this as I need to use directly (without the decorator) in some stories, as I need to change some settings

**Review by oisincoveney — APPROVED** [review]

File: e2e/support/helpers/e2e-embedding-helpers.js [hunk 8]
```diff
@@ -262,6 +262,12 @@ export function createPublicDashboardLink(dashboardId) {
   return cy.request("POST", `/api/dashboard/${dashboardId}/public_link`, {});
 }
 
+/**
+ * @param {Object} options
+ * @param {string} options.url
+ * @param {Object} options.qs
+ * @param {Function} [options.onBeforeLoad]
```
Comment (npretto) [inline e2e/support/helpers/e2e-embedding-helpers.js:269]: the function is optional, but typescript doesn't understand that without the jsdoc

**Review by oisincoveney — APPROVED** [review]
